src/test/java/com/team01/carsharingapp/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,333 @@
+import com.team01.carsharingapp.exception.EntityNotFoundException;
+import com.team01.carsharingapp.exception.RegistrationException;
+import com.team01.carsharingapp.mapper.UserMapper;
+import com.team01.carsharingapp.model.Role;
+import com.team01.carsharingapp.model.User;
+import com.team01.carsharingapp.repository.RoleRepository;
+import com.team01.carsharingapp.repository.UserRepository;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.AdditionalAnswers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceImplTest {
+    private static final Long ID_ONE = 1L;
+    private static final Long ID_TWO = 2L;
+    private static final String EMAIL = "email@example.com";
+    private static final String NEW_EMAIL = "new.email@example.com";
+    private static final String PASSWORD = "password";
+    private static final String NEW_PASSWORD = "newPassword";
+    private static final String HASHED_PASSWORD = "hashedPassword";
+    private static final String FIRST_NAME = "FirstName";
+    private static final String NEW_FIRST_NAME = "NewFirstName";
+    private static final String LAST_NAME = "LastName";
+    private static final String NEW_LAST_NAME = "NewLastName";
+
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private RoleRepository roleRepository;
+    @Mock
+    private PasswordEncoder passwordEncoder;
+    @Mock
+    private UserMapper userMapper;
+
+    @InjectMocks
+    private UserServiceImpl userServiceImpl;
+
+    @Test
+    @DisplayName("Register user with valid data")
+    void register_validData_returnUserRegistrationResponseDto() throws RegistrationException {
+        UserRegistrationRequestDto requestDto = getUserRegistrationRequestDto();
+        User user = getValidUser();
+
+        when(userMapper.toModel(requestDto)).thenReturn(user);
+        when(passwordEncoder.encode(PASSWORD)).thenReturn(HASHED_PASSWORD);
+        when(roleRepository.getRolesByName(Role.RoleName.ROLE_CUSTOMER)).thenReturn(new Role());
+        when(userRepository.save(user)).thenReturn(user);
+        when(userMapper.toResponseDto(user))
+                .thenReturn(getUserRegistrationResponseDtoFromUser(user));
+
+        UserRegistrationResponseDto expected = getUserRegistrationResponseDto();
+        UserRegistrationResponseDto actual = userServiceImpl.register(requestDto);
+
+        assertNotNull(actual);
+        assertEquals(expected, actual);
+        verify(userRepository, times(1)).save(user);
+        verify(userMapper, times(1)).toResponseDto(user);
```

Comment:
times(1) already set by default. You don't need to verify methods, if they use in when() method